src/components/svg/wrap/svg-keyboard-wrap.component.ts

```diff
@@ -0,0 +1,156 @@
+    state, animate, transition, trigger
+} from '@angular/core';
+
+import { KeyAction } from '../../../../config-serializer/config-items/KeyAction';
+import { SvgKeyboardComponent } from '../keyboard';
+import { PopoverComponent } from '../../popover';
+import { Module } from '../../../../config-serializer/config-items/Module';
+import { Layer } from '../../../../config-serializer/config-items/Layer';
+
+@Component({
+    selector: 'svg-keyboard-wrap',
+    template: require('./svg-keyboard-wrap.component.html'),
+    styles: [require('./svg-keyboard-wrap.component.scss')],
+    directives: [SvgKeyboardComponent, PopoverComponent],
+    animations: [
+        trigger('layerState', [
+            /* Right -> Left animation*/
+            state('leftIn', style({
+                transform: 'translateX(-50%)',
+                left: '50%'
+            })),
+            state('leftOut', style({
+                transform: 'translateX(-100%)',
+                left: '0'
+            })),
+            /* Right -> Left animation */
+            state('rightIn', style({
+                transform: 'translateX(-50%)',
+                left: '50%'
+            })),
+            state('rightOut', style({
+                transform: 'translateX(0%)',
+                left: '100%'
+            })),
+            /* Transitions */
+            transition('none => leftIn, leftOut => leftIn', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(0%)',
+                    left: '100%'
+                }),
+                style({
+                    opacity: 1
+                }),
+                animate('400ms ease-out')
+            ]),
+            transition('* => none', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(-100%)',
+                    left: '0'
+                }),
+                style({
+                    opacity: 1
+                })
+            ]),
+            transition('none => rightIn, rightOut => rightIn', [
+                style({
+                    opacity: 0,
+                    transform: 'translateX(-100%)',
+                    left: '0'
+                }),
+                style({
+                    opacity: 1
+                }),
+                animate('400ms ease-out')
+            ]),
+            transition(
+                'leftIn => leftOut,' +
+                'rightIn => rightOut,' +
+                'leftIn <=> rightOut,' +
+                'rightIn <=> leftOut',
+                animate('400ms ease-out')
+            )
+        ])
+    ]
+})
+export class SvgKeyboardWrapComponent implements OnInit {
+    @Input() layers: Layer[];
+    @Input() popoverEnabled: boolean = true;
+    @Input() animationEnabled: boolean = true;
+
+    private popoverShown: boolean;
+    private keyEditConfig: { moduleId: number, keyId: number };
+    private popoverInitKeyAction: KeyAction;
+    private module: Module[];
```

Comment:
Isn't `module` the same as `this.layers[this.currentLayer].modules.element` ? If it is, then I don't undestand why we should maintain a shortcut for it.